src/data/firebase.ts

```diff
@@ -0,0 +1,106 @@
+import { getImportedKey } from "./index";
+import * as firebase from "firebase/app";
+import "firebase/firestore";
+import { ExcalidrawElement } from "../element/types";
+import { getDrawingVersion } from "../element";
+
+const firebaseConfig = JSON.parse(process.env.REACT_APP_FIREBASE_CONFIG);
+
+firebase.initializeApp(firebaseConfig);
+
+interface FirebaseStoredDrawing {
+  drawingVersion: number;
+  ciphertext: firebase.firestore.Blob;
+}
+
+const EMPTY_IV = new Uint8Array(12);
+
+async function encryptElements(
+  key: string,
+  elements: readonly ExcalidrawElement[],
+): Promise<ArrayBuffer> {
+  const importedKey = await getImportedKey(key, "encrypt");
+  const iv = EMPTY_IV;
+  const json = JSON.stringify(elements);
+  const encoded = new TextEncoder().encode(json);
+  return await window.crypto.subtle.encrypt(
+    {
+      name: "AES-GCM",
+      iv,
+    },
+    importedKey,
+    encoded,
+  );
+}
+
+async function decryptElements(
+  key: string,
+  ciphertext: ArrayBuffer,
+): Promise<readonly ExcalidrawElement[]> {
+  const importedKey = await getImportedKey(key, "decrypt");
+  const decrypted = await window.crypto.subtle.decrypt(
+    {
+      name: "AES-GCM",
+      iv: EMPTY_IV,
+    },
+    importedKey,
+    ciphertext,
+  );
+
+  const decodedData = new TextDecoder("utf-8").decode(
+    new Uint8Array(decrypted) as any,
+  );
+  return JSON.parse(decodedData);
+}
```

Comment:
There's an almost identical code for the other encryption. Can you reuse it from there instead of duplicating it here?

Reply:
it's almost but not quite the same. i think we should roll this change out for a while and eventually kill `excalidraw-json` and all the code related to it in favor of having one way to do it on top of firebase. wdyt?

Reply:
Sounds good